src/widget/utils.js

```diff
@@ -52,6 +53,34 @@ export function widgetize( element ) {
 	return element;
 }
 
+/**
+ * Sets fake selection label for given element.
+ * It can be passed as a plain string or a function returning a string. Function will be called each time label is retrieved by
+ * {module:image/widget/utils~getFakeSelectionLabel}.
+ *
+ * @param {module:engine/view/element~Element} element
+ * @param {String|Function} labelOrCreator
+ */
+export function setFakeSelectionLabel( element, labelOrCreator ) {
+	element.setCustomProperty( fakeSelectionLabelSymbol, labelOrCreator );
+}
+
+/**
+ * Returns fake selection label for provided element.
+ *
+ * @param {module:engine/view/element~Element} element
+ * @return {String|undefined}
+ */
+export function getFakeSelectionLabel( element ) {
+	const labelCreator = element.getCustomProperty( fakeSelectionLabelSymbol );
+
+	if ( !labelCreator ) {
+		return undefined;
```

Comment:
An empty string would be better I think. Let's say that there's always some label but sometimes it's empty :).